rasa/nlu/registry.py

```diff
@@ -110,31 +120,41 @@ def get_component_class(component_name: Text) -> Type["Component"]:
         try:
             return rasa.shared.utils.common.class_from_module_path(component_name)
 
-        except AttributeError:
-            # when component_name is a path to a class but the path does not contain
-            # that class
-            module_name, _, class_name = component_name.rpartition(".")
-            raise Exception(
-                f"Failed to find class '{class_name}' in module '{module_name}'.\n"
-            )
-        except ImportError as e:
+        except (ImportError, AttributeError) as e:
             # when component_name is a path to a class but that path is invalid or
             # when component_name is a class name and not part of old_style_names
 
             is_path = "." in component_name
 
             if is_path:
-                module_name, _, _ = component_name.rpartition(".")
-                exception_message = f"Failed to find module '{module_name}'. \n{e}"
+                module_name, _, class_name = component_name.rpartition(".")
+                if isinstance(e, ImportError):
+                    exception_message = f"Failed to find module '{module_name}'."
+                else:
+                    # when component_name is a path to a class but the path does
+                    # not contain that class
+                    exception_message = (
+                        f"The class '{class_name}' could not be "
+                        f"found in module '{module_name}'."
+                    )
             else:
                 exception_message = (
-                    f"Cannot find class '{component_name}' from global namespace. "
+                    f"Cannot find class '{component_name}' in global namespace. "
                     f"Please check that there is no typo in the class "
                     f"name and that you have imported the class into the global "
                     f"namespace."
                 )
 
-            raise ModuleNotFoundError(exception_message)
+            raise ComponentNotFoundException(
+                f"Failed to load the component "
+                f"'{component_name}'. "
+                f"{exception_message} Either your "
+                f"pipeline configuration contain an error "
+                f"or the module you are trying to import "
+                f"is broken (e.g. the module is trying "
+                f"to import a package that is not "
+                f"installed). {traceback.format_exc()}"
```

Comment:
yes, there is a scenary where the stacktrace is very helpful: it might be the case that you configured the component correctly (using the right module path) but there is an importerror inside of your module, e.g. your class imports a package that does not exist. We can't figure this out on our own, so showing the stacktrace here will allow the user to see where in their custom code the error happend.